src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserReleaseRoleRepository.cs

```diff
@@ -176,6 +176,31 @@ public async Task<bool> IsUserEditorOrApproverOnLatestRelease(Guid userId, Guid
                 EditorAndApproverRoles);
         }
 
+        public async Task<bool> IsUserPrereleaseViewerOnLatestPreReleaseRelease(Guid userId, Guid publicationId)
+        {
+            var publication = await _contentDbContext.Publications
+                .Include(p => p.Releases)
+                .SingleAsync(p => p.Id == publicationId);
+
+            var latestRelease = publication.LatestRelease();
+
+            // Publication may have no releases
+            if (latestRelease == null
+                // Release should be in prerelease
+                || latestRelease.Published != null
+                || latestRelease.ApprovalStatus != ReleaseApprovalStatus.Approved)
+            {
+                return false;
+            }
```

Comment:
This doesn't necessarily check the release in prerelease as it's only checking there's an approved non-published latest release.

The `if` will evaluate `false` (i.e. the `return false` will be skipped over) if the latest release is approved but not published yet.

That means if they have the prerelease viewer role but prerelease hasn't started yet they will have access to the methodology.

I think I was expecting to see something either here to check the release is in prerelease or something in the calling authorization handler which is checking that and then the code here is just focussed on the role check.

Reply:
Added a check using `_preReleaseService.GetPreReleaseWindowStatus` and updated the unit tests.